Documentation for v5 mcsolve (#2042)
qutip/qutip: doc/guide/dynamics/dynamics-monte.rst

.. plot::
    :context:

>     from qutip.solver.mcsolve import MCSolver, mcsolve

Reviewer: ```suggestion
    from qutip.solver.mcsolve import mcsolve
```
I would suggest just importing it down when it is necessary.

Monte Carlo Solver Result
-------------------------

The Monte Carlo solver returns a :class:`qutip.MultitrajResult` object consisting of expectation values and/or states.

> The main difference with :func:`qutip.mesolve`'s :class:`qutip.Result` is that it optionally store the result of each trajectories with the averages.

Reviewer: ```suggestion
The main difference with :func:`qutip.mesolve`'s :class:`qutip.Result` is that it optionally stores the result of each trajectory together with their averages.
```

When trajectories are stored, ``result.runs_expect`` is a list over the expectation operators, trajectories and times in that order.
The averages are stored in ``result.average_expect`` and the standard derivation of the expectation values in ``result.std_expect``.

> When states are returned, ``result.runs_states`` will be an array of length ``ntraj``, with each element containing an array of ket-type qobjs with the same number of elements as ``times`` and ``result.average_states`` is a list of density matrices for each times.

Reviewer: ```suggestion
When the states are returned, ``result.runs_states`` will be an array of length ``ntraj``. Each element contains an array of "Qobj" type ket with the same number of elements as ``times``. ``result.average_states`` is a list of density matrices computed as the average of the states at each time step.
```

> Furthermore, the output will also contain a list of times at which collapse occurred, and which collapse operators did the collapse, in the ``col_times`` and ``col_which`` properties, respectively.

Reviewer: ```suggestion
Furthermore, the output will also contain a list of times at which the collapse occurred, and which collapse operators did the collapse. These can be obtained in  ``result.col_times`` and ``result.col_which`` respectively.
```

Lastly ``result.photocurrent`` contain the measurement of the evolution.

.. plot::
    :context: close-figs

    times = np.linspace(0.0, 10.0, 200)
    psi0 = tensor(fock(2, 0), fock(10, 8))
    a  = tensor(qeye(2), destroy(10))
    sm = tensor(destroy(2), qeye(10))
    H = 2*np.pi*a.dag()*a + 2*np.pi*sm.dag()*sm + 2*np.pi*0.25*(sm*a.dag() + sm.dag()*a)
    data = mcsolve(H, psi0, times, [np.sqrt(0.1) * a], e_ops=[a.dag() * a, sm.dag() * sm])

    plt.figure()
    plt.plot((times[:-1] + times[1:])/2, data.photocurrent[0])
    plt.title('Monte Carlo Photocurrent')
    plt.xlabel('Time')
    plt.ylabel('Photon detections')
    plt.show()

.. _monte-ntraj:

Changing the Number of Trajectories
-----------------------------------

> As mentioned earlier, by default, the ``mcsolve`` function runs 500 trajectories.

Reviewer: ```suggestion
By default, the ``mcsolve`` function runs 500 trajectories.
```
I think the paragraph that mentions mcsolve has by default 500 trajectories was deleted.

This value was chosen because it gives good accuracy, Monte Carlo errors scale as :math:`1/n` where :math:`n` is the number of trajectories, and simultaneously does not take an excessive amount of time to run.

> However, like many other options in QuTiP you are free to change the number of trajectories to fit your needs.

Reviewer: ```suggestion
However, you can change the number of trajectories to fit your needs.
```

> If we want to run 1000 trajectories in the above example, we can simply modify the call to ``mcsolve`` like:

Reviewer: ```suggestion
In order to run 1000 trajectories in the above example, we can simply modify the call to ``mcsolve`` like:
```
